src/app/controller/services/base/port.service.ts

```diff
@@ -46,12 +48,17 @@ export class PortService {
     );
   }
 
-  getPorts = (): Observable<Port[]> => this.httpClient.get<Port[]>(this.PORT_URL_LIMIT_1000)
-    .pipe(map(ports => {
-      for (let port of ports) {
-        cachePort(this.unlocode2PortCache, port);
-      }
-      return ports;
-    }));
-
+  getPorts(): Observable<Port[]> {
+    if (!this.ports$) {
+      this.ports$ = this.httpClient.get<Port[]>(this.PORT_URL_LIMIT_1000).pipe(map(ports => {
+        for (let port of ports) {
+          cachePort(this.unlocode2PortCache, port);
+        } 
+        return ports;
+      }),
```

Comment:
This should probably have been a `tap` rather than a `map`

Reply:
I agree